### PR TITLE
Fix trajectory_predict_skypos mutating input array

### PR DESCRIPTION
**Pull request by wilsonbb**
`trajectory_predict_skypos` is meant to predict the positions of a KBMOD trajectory along a list or array of observation times, however it mutates the observation times if they are provided as a numpy array rather than a list.

It calls `dt = np.asarray(times)` which creates a new numpy array if `times` is a list or is a no-op if it is an array. We then subtract out the starting timestamp so all times are offset from 0, and we use the in-place operation of `dt -= dt[0]` to preserve memory. However if `times` is an array, the original object passed to the array has been mutated to now be offset by 0.

As quickly illustrated in a python repl:

```
>>> import numpy as np
>>> def modify_me(n):
...     new_n = np.asarray(n)
...     new_n -= new_n[0]
... 
>>> a = list(range(10, 20))
>>> modify_me(a)
>>> a
[10, 11, 12, 13, 14, 15, 16, 17, 18, 19]
>>> b = np.asarray(a)
>>> modify_me(b)
>>> b
array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9]) 
```
This PR fixes this by choosing not to do the in-place subtraction.

To add better testing for this, we a) use tests with non-zero observation times to actually test the subtraction and b) test both lists and numpy arrays as inputs.

Note that this was showing up as an issue with the `KnownObjMatcher` having its numpy array of observation times mutated by the call to `trajectory_predict_skypos`. Then when it tries to compare the observation times against times from an ephemeris, a match cannot be made due to the temporal subtraction.

### CODE REVIEW

File: src/kbmod/trajectory_utils.py [hunk 1]
```diff
@@ -124,7 +124,9 @@ def trajectory_predict_skypos(trj, wcs, times):
         A SkyCoord with the transformed locations.
     """
     dt = np.asarray(times)
-    dt -= dt[0]
+    # Note that we do a reassignment to avoid modifying the input, which may
+    # happen if `times` is already an array and `np.asarray` is a no-op.
+    dt = dt - dt[0]
```
Comment (jeremykubica) [inline src/kbmod/trajectory_utils.py:129]: Nice fix. It might be slightly clearer if we also rename it to something like `zeroed_dt`, but I leave that up to you.

File: tests/test_known_object_filters.py [hunk 2]
```diff
@@ -107,6 +110,9 @@ def setUp(self):
             time_offset=time_offset_mjd_close,
         )
 
+        # check that the obstimes have been unmodified and are still not zero-offset
+        self.assertTrue(np.all(self.obstimes >= start_time))
```
Comment (jeremykubica) [inline tests/test_known_object_filters.py:114]: To make sure they are completely unmodified, I think it is also worth adding `times_copy = np.copy(self.obstimes)` around line 30 and `self.assertTrue(np.all(self.obstimes == times_copy))` or something similar.

File: tests/test_trajectory_utils.py [hunk 3]
```diff
@@ -62,12 +62,19 @@ def test_predict_skypos(self):
...
-        self.assertAlmostEqual(my_sky.dec[0].deg, -15.0)
-        self.assertAlmostEqual(my_sky.ra[1].deg, 45.2, delta=0.01)
-        self.assertAlmostEqual(my_sky.dec[1].deg, -15.5, delta=0.01)
+        # Predict locations at times 57921.0 and 57922.0, both as a list and as a numpy array.
+        obstimes = [57921.0, 57922.0]
+        for curr_obstimes in [obstimes, np.array(obstimes)]:
+            my_sky = trajectory_predict_skypos(trj, my_wcs, curr_obstimes)
+            # Verify that the obstimes were not mutated
+            self.assertEqual(curr_obstimes[0], 57921.0)
+            self.assertEqual(curr_obstimes[1], 57922.0)
```
Comment (jeremykubica) [inline tests/test_trajectory_utils.py:71]: Nice test!

**Review by jeremykubica — APPROVED** [review]
Looks good. I added some optional comments (one stylistic and one extending the test).

File: tests/test_known_object_filters.py (hunk 2 above)
Comment (wilsonbb) [reply tests/test_known_object_filters.py:115]: Ah good catch thanks!

File: src/kbmod/trajectory_utils.py (hunk 1 above)
Comment (wilsonbb) [reply src/kbmod/trajectory_utils.py]: Good, call
